### PR TITLE
Refactored parseNumber scenario

### PR DESCRIPTION
**Pull request by MikhailAlferov**

### CODE REVIEW

File: src/Client/InfoObtainingClient/InfoClient.php [hunk 1]
```diff
@@ -184,6 +186,94 @@ public function getAllChats(callable $onComplete): void {
...
+        $this->contactsKeeper->addNumbers($numbers, $onComplete);
+    }
+
+    /**
+     * @param array    $numbers
+     * @param callable $onComplete
+     */
+    public function delNumbers(array $numbers, callable $onComplete)
+    {
+        $this->contactsKeeper->delNumbers($numbers, function () use ($onComplete) {
```
Comment (wkejqb) [inline src/Client/InfoObtainingClient/InfoClient.php]: `$onComplete` can be passed directly, without wrapper function.

File: src/Client/InfoObtainingClient/InfoClient.php [hunk 2]
```diff
@@ -184,6 +186,94 @@ public function getAllChats(callable $onComplete): void {
...
+            } else {
+                $addNumbersFunc();
+            }
+        });
+    }
+
+    /**
+     * @param callable $onComplete
+     */
+    public function cleanContacts(callable $onComplete)
```
Comment (wkejqb) [inline src/Client/InfoObtainingClient/InfoClient.php]: can we just add `getContactsKeeper()` method, and call contact keeper directly?

File: src/Scenario/MyTgClientDebug.php [hunk 3]
```diff
@@ -0,0 +1,393 @@
+<?php
+
+namespace TelegramOSINT\Scenario;
+
+use TelegramOSINT\Client\AuthKey\AuthKeyCreator;
+use TelegramOSINT\Client\Client;
+use TelegramOSINT\Client\InfoObtainingClient\InfoClient;
+use TelegramOSINT\Client\InfoObtainingClient\Models\UserInfoModel;
+use TelegramOSINT\Client\StatusWatcherClient\Models\HiddenStatus;
```
Comment (anton-postuf) [inline src/Scenario/MyTgClientDebug.php]: This file is obsolete

File: src/Scenario/UserContactsScenario.php [hunk 4]
```diff
@@ -61,42 +62,58 @@ public function parseNumbers(array $numbers, bool $withPhoto = false, bool $larg
...
-                        }
-                    } else {
-                        $counter--;
-                    }
+                        });
+                    });
+                    sleep(2);
+                }
+            });
+        };
```
Comment (anton-postuf) [inline src/Scenario/UserContactsScenario.php]: Alghorithm should be following:
1) reloadContacts
2) rememberedContacts = getCurrentContacts (get all possible info, except first/last names)
3) cleanContacts
4) foreach contact in rememberedContacts { getUserFull(contact) -> setuo last\first names }

File: examples/parseNumbers.php [hunk 5]
```diff
@@ -17,31 +17,37 @@
...
+    $photo_file = '';
+    if ($model->photo){
+        $photo_file = $model->phone.'.'.$model->photo->format;
+        file_put_contents(
+            $photo_file,
+            $model->photo->bytes
+        );
     }
-});
+    echo $model->phone."\t|\t".
```
Comment (wkejqb) [inline examples/parseNumbers.php]:
```
implode("\t|\t", [
   ...
]);
```
would look nicer.

**Pull request merged**